### ISSUE TITLE
Creating stochastic fine-tuning test

### ISSUE THREAD

**Issue by alessandramaranca** [niznik-dev/predicting-zygosity#5]
This is a stochastic test with parameters _p_ (probability of 1 or 0 as outcome) and _n_ (number of bits), (@msalganik's idea). Generate every possible sequence of _n_ bits (for instance, for _n=3 it_ would be 001, 101, 000, 111, 100, 010, 011, 110). With probability _p_, assign to each sequence 1 with probability _p_ (and 0 with probability _1-p_). We can do these in the same format we are generating books (for instance, for n=7 we could have {input = {0101011}, output = 1}. Have no duplicate sequences, and do a test/train split (do a fixed sample size instead of a percentage). If _p_=0.5 we would expect the training loss going to 0 (the LLM memorizes the entire sequence) with enough epochs, and we would expect it perform exactly according to the random threshold _p = 0.5_ in the test set (since there's nothing to learn). This would be a test of whether it can memorize a simple signal in the training set and fully learn from our training set, in addition to test whether there is leakage (we see that if it performs above the random threshold _p_ in the test set, something is wrong and it might be indication of leakage). Moreover, if it performs significantly below the random threshold _p_, something might be wrong with the fine-tuning. 

**Steps in creating this test are:**
1. Create sequence books like specified above. 
2. Do train/test split.
3. Finetune on train set. 
4. Evaluate on test set. 
5. Check whether accuracy is approximately equal to _p_.

**Comment by niznik-dev** [comment]
@alessandramaranca If you haven't made a branch for this yet, great! Because I just merged the last outstanding thing on my end into main so you'll be all set in terms of recent major changes.

If you did, a reminder that you'll probably want to merge the changes from main back into your branch - a decent amount has changed from when you first started coding this.

**Comment by alessandramaranca** [comment]
> [@alessandramaranca](https://github.com/alessandramaranca) If you haven't made a branch for this yet, great! Because I just merged the last outstanding thing on my end into main so you'll be all set in terms of recent major changes.
> 
> If you did, a reminder that you'll probably want to merge the changes from main back into your branch - a decent amount has changed from when you first started coding this.

Made a draft branch and added it! Still need to fix the paths to make sure they are pointing to the right places.
